<?xml version="1.0" encoding="UTF-8"?>
<launch>
	<arg name="sigverse_ros_bridge_port"    default="50001" />
	<arg name="ros_bridge_port"             default="9090" />
	<arg name="camera"                      default="/camera" />
	
	<node pkg="sigverse_turtlebot3_open_manipulator" exec="turtlebot3_teleop_key" output="screen" launch-prefix="gnome-terminal -t 'TurtleBot3 teleop key' -- ">
		<!--
		<remap from="/tb3/joint_state"         to="/tb3/joint_state" />
		<remap from="/tb3/cmd_vel"             to="/tb3/cmd_vel" />
		<remap from="/tb3/joint_trajectory"    to="/tb3/joint_trajectory" />
		-->
	</node>


	<node pkg="sigverse_ros_bridge" exec="sigverse_ros_bridge" args="$(var sigverse_ros_bridge_port)"/>

	<include file="$(find-pkg-share rosbridge_server)/launch/rosbridge_websocket_launch.xml" > 
		<arg name="port" value="$(var ros_bridge_port)"/>
		<arg name="default_call_service_timeout" value="5.0"/>
		<arg name="call_services_in_new_thread" value="true"/>
		<arg name="send_action_goals_in_new_thread" value="true"/>
		<!-- <arg name="bson_only_mode" value="true"/> -->
	</include>

	<node pkg="image_tools" exec="showimage" name="showimage">
		<remap from="image" to="$(var camera)/rgb/image_raw"/>
	</node>
</launch>

